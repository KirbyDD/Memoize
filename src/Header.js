import React, { Component } from "react";
import "./Header.scss";

class Header extends Component {
  constructor() {
    super();
    this.state = {};
  }
  render() {
    return (
      <header>
        <div className="gun-logo">
          <img
            src="https://www.aeroprecisionusa.com/media/catalog/product/cache/8a4e8be590f5e34deaab5118a0859ee5/a/p/aprh100033-m4-sliding-stock-1_1.jpg"
            className="logo-stock"
          />
          <h1>Scope Practice</h1>
          <img
            src="http://tradingplacepawn.com/media/catalog/product/cache/1/image/9df78eab33525d08d6e5fb8d27136e95/l/a/lantac-dragon-3.jpg"
            className="logo-muzzle"
          />
        </div>
        <img
          src="https://image.sportsmansguide.com/adimgs/l/reticle_445_ts.jpg"
          alt="Weapon scope crosshair"
        />
      </header>
    );
  }
}

export default Header;
