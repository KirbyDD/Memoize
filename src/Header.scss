header {
  align-items: center;
  background-color: white;
  box-shadow: 5px 5px 10px;
  display: flex;
  justify-content: space-around;
  margin: 40px auto 50px;
  width: 80%;
  img {
    height: 100px;
    width: 100px;
  }
}

.gun-logo {
  align-items: center;
  display: flex;
}

.logo-stock{
  padding-top: 50px;
}

.logo-muzzle{
  padding-top: 10px;
}

h1 {
  font-size: 60px;
}