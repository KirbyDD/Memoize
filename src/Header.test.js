import React from 'react';
import Header from './Header.js';
import { shallow } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

describe('CardContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <Header />
    )
  })
  it('should match snapshot when data is passed in ', () => {
    expect(wrapper).toMatchSnapshot();
  });
})