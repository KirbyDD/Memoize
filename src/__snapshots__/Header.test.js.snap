// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardContainer should match snapshot when data is passed in  1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="gun-logo"
        >
          <img
            className="logo-stock"
            src="https://www.aeroprecisionusa.com/media/catalog/product/cache/8a4e8be590f5e34deaab5118a0859ee5/a/p/aprh100033-m4-sliding-stock-1_1.jpg"
          />
          <h1>
            Scope Practice
          </h1>
          <img
            className="logo-muzzle"
            src="http://tradingplacepawn.com/media/catalog/product/cache/1/image/9df78eab33525d08d6e5fb8d27136e95/l/a/lantac-dragon-3.jpg"
          />
        </div>,
        <img
          alt="Weapon scope crosshair"
          src="https://image.sportsmansguide.com/adimgs/l/reticle_445_ts.jpg"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="logo-stock"
              src="https://www.aeroprecisionusa.com/media/catalog/product/cache/8a4e8be590f5e34deaab5118a0859ee5/a/p/aprh100033-m4-sliding-stock-1_1.jpg"
            />,
            <h1>
              Scope Practice
            </h1>,
            <img
              className="logo-muzzle"
              src="http://tradingplacepawn.com/media/catalog/product/cache/1/image/9df78eab33525d08d6e5fb8d27136e95/l/a/lantac-dragon-3.jpg"
            />,
          ],
          "className": "gun-logo",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "logo-stock",
              "src": "https://www.aeroprecisionusa.com/media/catalog/product/cache/8a4e8be590f5e34deaab5118a0859ee5/a/p/aprh100033-m4-sliding-stock-1_1.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Scope Practice",
            },
            "ref": null,
            "rendered": "Scope Practice",
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "logo-muzzle",
              "src": "http://tradingplacepawn.com/media/catalog/product/cache/1/image/9df78eab33525d08d6e5fb8d27136e95/l/a/lantac-dragon-3.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Weapon scope crosshair",
          "src": "https://image.sportsmansguide.com/adimgs/l/reticle_445_ts.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="gun-logo"
          >
            <img
              className="logo-stock"
              src="https://www.aeroprecisionusa.com/media/catalog/product/cache/8a4e8be590f5e34deaab5118a0859ee5/a/p/aprh100033-m4-sliding-stock-1_1.jpg"
            />
            <h1>
              Scope Practice
            </h1>
            <img
              className="logo-muzzle"
              src="http://tradingplacepawn.com/media/catalog/product/cache/1/image/9df78eab33525d08d6e5fb8d27136e95/l/a/lantac-dragon-3.jpg"
            />
          </div>,
          <img
            alt="Weapon scope crosshair"
            src="https://image.sportsmansguide.com/adimgs/l/reticle_445_ts.jpg"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                className="logo-stock"
                src="https://www.aeroprecisionusa.com/media/catalog/product/cache/8a4e8be590f5e34deaab5118a0859ee5/a/p/aprh100033-m4-sliding-stock-1_1.jpg"
              />,
              <h1>
                Scope Practice
              </h1>,
              <img
                className="logo-muzzle"
                src="http://tradingplacepawn.com/media/catalog/product/cache/1/image/9df78eab33525d08d6e5fb8d27136e95/l/a/lantac-dragon-3.jpg"
              />,
            ],
            "className": "gun-logo",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "logo-stock",
                "src": "https://www.aeroprecisionusa.com/media/catalog/product/cache/8a4e8be590f5e34deaab5118a0859ee5/a/p/aprh100033-m4-sliding-stock-1_1.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Scope Practice",
              },
              "ref": null,
              "rendered": "Scope Practice",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "logo-muzzle",
                "src": "http://tradingplacepawn.com/media/catalog/product/cache/1/image/9df78eab33525d08d6e5fb8d27136e95/l/a/lantac-dragon-3.jpg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Weapon scope crosshair",
            "src": "https://image.sportsmansguide.com/adimgs/l/reticle_445_ts.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
